src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(getFormattedData(dateFrom, "-", dateTo));
+        for (String name : names) {
+            int sumSalary = 0;
+            for (String value : data) {
+                String[] dataSplitted = value.split(" ");
+                if (isWithinPeriod(dataSplitted[0], dateFrom, dateTo)
+                        && name.equals(dataSplitted[1])) {
+                    sumSalary += Integer.parseInt(dataSplitted[2])
+                            * Integer.parseInt(dataSplitted[3]);
+                }
+            }
+            result.append(System.lineSeparator())
+                    .append(getFormattedData(name, "-", Integer.toString(sumSalary)));
+        }
+        return result.toString();
+    }
+
+    private String getFormattedData(String...data) {
+        return String.join(" ", data);
+    }
+
+    private boolean isWithinPeriod(String data, String dataFrom, String dataTo) {
+        return !encodeData(data).isBefore(encodeData(dataFrom))
+                && !encodeData(data).isAfter(encodeData(dataTo));
```

Comment:
The method 'encodeData' is called multiple times with the same arguments within 'isWithinPeriod'. This is inefficient as it parses the same date strings repeatedly. Consider parsing 'dataFrom' and 'dataTo' once outside of the 'isWithinPeriod' method and passing the resulting LocalDate objects to it instead.